Added required shared docs and updated old docs (#213)
exercism/purescript: docs/TESTS.md

# Tests

The PureScript exercise test suites may be run from the exercise directory
using the spago tool:

>     cd <purescript-exercism>/exercises/<exercise-type>/<exercise name>
>     spago test
> 
> where <purescript-exercism> is the root directory where you checked out this
> repository.

Reviewer: Students don't checkout this repo but work with exercises via the Exercism CLI. As such, this path is not referring to the repository's directory but the directory on a student's machine when using the exercism CLI. 

```suggestion
    cd ~/<exercism-workspace>/purescript/<exercise name>
    spago test

where <exercism-workspace> is the CLI's workspace directory.
```

Author: I'll update.

Author: When students check out, will they have `~/<exercism-workspace>/purescript/<exercise name>` or `~/<exercism-workspace>/<exercise name>`? I.e. is `purescript` part of the folder structure?

Reviewer: `purescript` is part of the folder structure

For example:

>     cd ~/purescript/purescript-exercism/exercises/practice/accumulate

Reviewer: ```suggestion
    cd ~/exercism/purescript/accumulate
```

Author: I'll update.
